src/wp-admin/options-general.php

```diff
@@ -506,6 +506,34 @@
 ?>
 </select></td>
 </tr>
+
+
+<tr>
+<th scope="row"><label for="blocks_compatibility_level"><?php _e( 'Blocks Compatibility' ); ?></label></th>
+<td><select name="blocks_compatibility_level" id="blocks_compatibility_level">
+<?php
+$blocks_compatibility_level = get_option( 'blocks_compatibility_level', 1 );
+$blocks_compatibility_level_desc = array(
+	__( 'Off' ),
+	__( 'On' ),
+	__( 'Troubleshooting' ),
+);
+for ( $index = 0; $index <= 2; $index++ ) :
+	$selected = ( $blocks_compatibility_level == $index ) ? 'selected="selected"' : '';
+	echo "\n\t<option value='" . esc_attr( $index ) . "' $selected>" . $blocks_compatibility_level_desc[ $index ] . '</option>';
```

Comment:
We could used the core `selected()` function here, maybe also some spacing and for the `foreach` I think we should use `{}`.